include/taskolib/GitObjectWrapper.h

```diff
@@ -0,0 +1,106 @@
+/**
+ * \file   GitObjectWrapper.h
+ * \author Sven-Jannik Wöhnert
+ * \date   Created on March 20, 2023
+ * \brief  Wrapper for data types defined in libgit2
+ *
+ * \copyright Copyright 2023 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+
+#include <git2.h>
+#include <utility>
+#include <typeinfo>
+#include <string>
+
+
+namespace task {
+
+// free C-type pointer (overload)
+void free_libgit_ptr(git_tree* tree);
+void free_libgit_ptr(git_signature* signature);
+void free_libgit_ptr(git_index* index);
+void free_libgit_ptr(git_repository* repo);
+void free_libgit_ptr(git_remote* remote);
+
+
+/**
+ * Template class wrapper for all libgit2 pointer types starting with git_{a-z}*.
+ * Copy methods are excluded to prevent double ownership of said pointer.
+ * 
+ * \class T: types like git_remote*, git_tree*, git_commit*, ...
```

Comment:
Doxygen uses `\tparam` to document template parameters:
```suggestion
 * \tparam T   a pointer type like git_remote*, git_tree*, git_commit*, ...
```